lib/src/install.rs

```diff
@@ -1344,9 +1357,11 @@ async fn install_to_filesystem_impl(state: &State, rootfs: &mut RootSetup) -> Re
     tracing::debug!("bound images={bound_images:?}");
 
     // Verify each bound image is present in the container storage
-    let bound_images = {
+    let resolved_bound_images = if state.config_opts.pull {
+        Vec::new()
```

Comment:
It feels to me like this is more like `Option<Vec<T>>` or so - basically we're either pulling the images or we aren't, i.e. we either have a complete vector or not.

---

Tangentially related one thing that might be a nice followup is that we add the host container storage as a read-only additional image store; I think this would get us "pull if not present" semantics.

Reply:
>It feels to me like this is more like Option<Vec<T>> or so - basically we're either pulling the images or we aren't, i.e. we either have a complete vector or not.

Indeed, that was my original implementation but it felt a bit (subjectively) messy

Reply:
Note that AIS (additional image store) is just a lookaside cache. In that case we'd still query the registry for the manifest+config at least, but reuse any locally stored layers.

So it'd still be pulling. 

That's distinct from what we do today without `--pull` for LBIs where if you have a tagged image we don't query the registry, we just take whatever we find in `containers-storage:`.

Reply:
I see, created followup ticket #863

As for the `Option` thing, I've implemented a 3-way enum. I tried to use `Option<Either<..,..>>` but it was gross